flink-streaming-java/src/main/java/org/apache/flink/streaming/api/operators/Triggerable.java

```diff
@@ -15,38 +15,26 @@
  * See the License for the specific language governing permissions and
  * limitations under the License.
  */
+package org.apache.flink.streaming.api.operators;
 
-package org.apache.flink.streaming.runtime.operators.windowing;
+import org.apache.flink.annotation.Internal;
 
-import org.apache.flink.streaming.runtime.operators.Triggerable;
-import org.apache.flink.streaming.runtime.tasks.TimeServiceProvider;
-
-import java.util.concurrent.ScheduledFuture;
-
-class NoOpTimerService extends TimeServiceProvider {
-
-	private volatile boolean terminated;
-
-	@Override
-	public long getCurrentProcessingTime() {
-		return System.currentTimeMillis();
-	}
-
-	@Override
-	public ScheduledFuture<?> registerTimer(long timestamp, Triggerable target) {
-		return null;
-	}
-
-	@Override
-	public boolean isTerminated() {
-		return terminated;
-	}
-
-	@Override
-	public void quiesceAndAwaitPending() {}
-
-	@Override
-	public void shutdownService() {
-		terminated = true;
-	}
+/**
+ * Interface for things that can be called by {@link InternalTimerService}.
+ *
+ * @param <K> Type of the keys to which timers are scoped.
+ * @param <N> Type of the namespace to which timers are scoped.
+ */
+@Internal
+public interface Triggerable<K, N> {
```

Comment:
Done, changed it to `ProcessingTimeCallback`, which has a method `onProcessingTime()`.